src/standalone/api/kernels/backgroundExecution.ts

```diff
@@ -94,31 +94,33 @@ del __jupyter_exec_background__
             );
         })
         // We no longer need to track any more outputs from the kernel that are related to this output.
-    ).finally(() => kernel.session && unTrackDisplayDataForExtension(kernel.session, displayId));
+    ).finally(() => {
+        kernel.session && unTrackDisplayDataForExtension(kernel.session, displayId);
+        disposables.dispose();
+    });
 
     for await (const output of api.executeCode(codeToSend, token)) {
         if (token.isCancellationRequested) {
             return;
         }
-        const error = output.items.find((item) => item.mime === mimeErrorResult);
```

Comment:
This isn't right
if the bg thread throws an error, then we sent `mimErrorResult`. (see line `60`).
Thus in the past when bg thread fails, we get the error and stop waiting for output and return.
But now we will never resolve this promise because we never get an output back agaihn.